plugins/lookup/secretsmanager_random_password.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Andrew Roth <andrew@andrewjroth.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+name: secretsmanager_random_password
+author:
+  - Andrew Roth (@andrewjroth) <andrew@andrewjroth.com>
+
+short_description: Generate a random password using AWS Secrets Manager
+description:
+  - Look up (really generate) a random password using AWS Secrets Manager's
+    `secretsmanager:GetRandomPassword` API.
+  - Optional parameters can be passed into this lookup; I(password_length) and I(exclude_characters)
+
+options:
+  _terms:
+    description: As a shortcut, the password_length parameter can be specified as a term instead of using the keyword.
+    required: False
+    type: integer
+  password_length:
+    description: The length of the password. If you do not include this parameter, the default length is 32 characters.
+    required: False
+    type: integer
```

Comment:
With the current behaviour where the length can also be passed as an unnamed parameter, the description isn't quite true, additionally, the 32 might change if Amazon decides to update their defaults.  If you explicitly set a default here, then at least the integration tests won't break should Amazon decide to update their defaults.

Reply:
I've removed the length check from the integration test when using the default for length and updated the documentation to be clear about this behavior.